x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -234,18 +234,20 @@ public static DataType commonType(DataType left, DataType right) {
         return DataTypeConverter.commonType(left, right);
     }
 
+    // generally supporting abbreviations from https://en.wikipedia.org/wiki/Unit_of_time
```

Comment:
Then we should also add smaller time units, like one [Jiffy](https://en.wikipedia.org/wiki/Unit_of_time#Scientific) and the Planck time. /s